src/bungie_api.py

```diff
@@ -45,9 +45,12 @@ def get_primary_membership_id_and_type(self, username):
                 ]
                 == membership_type
             ):
-                print(f"Crosave override found for {membership_id}")
-                return membership_id, membership_type
 
+                print(f"Cross-save override found for {membership_id}")
+            
+            return membership_id, membership_type
```

Comment:
Moving the return here will have it return the first `player` in the response.  I'm guessing that works for your friend because they only have a single account and I think it might be luck that it works for you and that your PSN account (which is cross save) comes before your XBox one.  I tested it on my account and it does work for me too, but I'm not sure that's guaranteed.

If there's only one `player` in the response, I could return that.  I think that might work for your friend.

There's an edge case here that I know I'm not handling where someone didn't use cross save but has multiple accounts.